internal/workflow/processing.go

```diff
@@ -760,6 +761,10 @@ func (w *ProcessingWorkflow) SessionHandler(
 				return err
 			}
 		}
+
+		if err := w.poststorage(sessCtx, tinfo.SIPID); err != nil {
```

Comment:
I think the AIS poststorage workflow is just going to fail if we aren't using AM for preservation right now. I think you either need to leave this out, or add some sort of escape path in the AIS workflow if a3m is the configured preservation system.

Reply:
I'm not sure I get your point. If they are not using AM it will probably fail before reaching this point for them. But we still want to have the option to run poststorage child workflows after a3m and storage as a general solution. Then the child workflow could use Enduro's storage API to fetch the AIP, instead of AMSS as the AIS workflow does.

The child workflow must be developed based on the custom needs, because they will only use AM we don't need to consider a3m at this point in the AIS workflow. Which would require a different implementation, as Enduro doesn't have an extract file endpoint.

Reply:
Okay, but if a system administrator configured Enduro to run the AIS poststorage workflow and forgot to configure AM as the preservation system, then the poststorage workflow would error out trying to create the AIS package. Because there's no reporting on poststorage events in the Dashboard the operator would have to look at the Temporal UI to try and figure out why AIS packages aren't being delivered.

Reply:
Sure, we may want to add some information about them in the API/UI in the future. But I don't see that as a reason to remove this call or to change the AIS child workflow.

Reply:
My reservations come from a few things:

1. The future of the Enduro with a3m is uncertain. We may never exercise the code in the a3m branch. It seems wrong to add code that is not used or tested.
2. The poststorage child workflow has no way to know which preservation system Enduro is using. This seems like a  problem when implementing poststorage workflows that depend on one particular preservation system.
3. Having two call points for the same poststorage workflow where one works and the other fails seems like a potential source for weird and confusing bugs.

I don't understand your argument that "If they are not using AM it will probably fail before reaching this point for them."  I think the a3m workflow should work fine up until the poststorage call (I may be wrong , I haven't tested this branch in a while).  At this point, if Enduro is configured to run with a3m and a poststorage workflow is configured to run that only works with AM, then this call will return an error.

Reply:
Welcome back @djjuhasz!

1. I was just talking about this with @aseles13, for now the a3m branch is still there, so we have to consider it. Not for them, but as a general solution. Same as with [the PREMIS validation work](https://github.com/artefactual-sdps/enduro/pull/1023#discussion_r1819505549).
2. Child workflows are a way to extend the workflow with custom activities based on the user needs. If the user needs to consider both preservation/storage systems, the child workflow should consider that too. Because both things are configurable, you can have a poststorage child workflow that only works with AM.
3. Same as above, if you are using a3m in one Enduro instance and AM in the other, link each of them with their corresponding child workflow (or make the child workflow work with both).
4. I test the a3m workflow branch from time to time, if you enable their preprocessing child workflow, a3m fails on "Verify metadata directory checksums", if you don't use it it works (tested with `little_vecteur_aip.zip`). Again, I'd consider a configuration issue.

Reply:
Okay. I'd prefer to have a way to have a poststorage workflow to exit cleanly (without an error) if it only works with AM, and Enduro is configured for a3m, rather than relying on correct configuration, but you obviously disagree. I'll approve the PR.